core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -268,6 +270,86 @@ def validate_leading_factor_of_delay(
         ]
 
 
+def validate_funding_spent(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """Validates that total reported funding spent on the sheet is less than the amount specified as allocated in a
+    separate sheet stored on S3.
+
+    For Town deal funds this is done on a per-project basis whereas for future High Street Funds this is done
+    per-programme leading to slightly different logics for both.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors
+    """
+    # Pull programme and project indexes from the workbook
+    programme_id = workbook["Programme_Ref"]["Programme ID"][0]
+    fund_type = programme_id[:2]
+    project_ids = workbook["Project Details"]["Project ID"]
+    funding_df = workbook["Funding"]
+
+    # pull funding spent for individual projects and programme wide total
+    funding_spent = {project: spend_per_project(funding_df, project) for project in project_ids}
+    funding_spent[programme_id] = sum(funding_spent.values())
+
+    allocated_file = s3.get_file("data-store-file-assets-dev", f"{fund_type}-grant-awarded.csv")
+    funding_allocated = pd.read_csv(allocated_file, index_col="Index Code")["Grant Awarded"]
```

Comment:
Can we just do this once during config instantiation and then use it here

Reply:
We decided to load this on validation instead as we didn't want potentially dynamic functions in const.py